docs/shipping/Code/go.md

```diff
@@ -245,239 +248,250 @@ Add the exporter definition to your application code:
-    metric.WithDescription("int_counter description"),
-)
-floatCounter := metric.Must(meter).NewFloat64Counter(
-    "go_metrics.float_counter",
-    metric.WithDescription("float_counter description"),
+counter, err := meter.Int64Counter(
+    "int_counter",  // name of the metric
+    // m.WithUnit("ms")  // add unit to the metric if you want
+    m.WithDescription("Counts the total number of requests"),
 )
 
+floatCounter, err := meter.Float64Counter("float_counter")  // name of the metric
+
 // Record values to the metric instruments and add labels
-intCounter.Add(ctx, int64(10), attribute.String("<<LABEL_KEY>>", "<<LABEL_VALUE>>"))
-floatCounter.Add(ctx, float64(2.5), attribute.String("<<LABEL_KEY>>", "<<LABEL_VALUE>>"))
+counter.Add(con, 1)
+counter.Add(con, 10, m.WithAttributes(attribute.Key("<<LABEL_KEY>>").String("<<LABEL_VALUE>>")))
+floatCounter.Add(con, 2.5)
 ```
 
 #### Asynchronous Counter
 
 ```go
-// Use `meter` from last steps.
-
-// Create callbacks for your CounterObserver instruments
-intCounterObserverCallback := func(_ context.Context, result metric.Int64ObserverResult) {
-    result.Observe(10, attribute.String("<<LABEL_KEY>>", "<<LABEL_VALUE>>"))
-}
-floatCounterObserverCallback := func(_ context.Context, result metric.Float64ObserverResult) {
-    result.Observe(2.5, attribute.String("<<LABEL_KEY>>", "<<LABEL_VALUE>>"))
-}
-
-// Create CounterObserver instruments
-_ = metric.Must(meter).NewInt64CounterObserver(
-    "go_metrics.int_counter_observer",
-    intCounterObserverCallback,
-    metric.WithDescription("int_counter_observer description"),
+// Create callbacks for your observable counter instruments
+// Float64ObservableCounter is also a supported type
+observableCounter, err := meter.Int64ObservableCounter(
+    "observable_counter",  // name of the metric
+    // m.WithUnit("By"),  // add unit to the metric if you want
+    m.WithDescription("observable counter description"),
 )
-_ = metric.Must(meter).NewFloat64CounterObserver(
-    "go_metrics.float_counter_observer",
-    floatCounterObserverCallback,
-    metric.WithDescription("float_counter_observer description"),
+
+_, err = meter.RegisterCallback(
+    func(con context.Context, o m.Observer) error {
+        o.ObserveInt64(observableCounter, 10, m.WithAttributes(attribute.Key("<<LABEL_KEY>>").String("<<LABEL_VALUE>>")))
+        return nil
+    },
+    observableCounter,
 )
 ```
 
 #### Histogram
 
 ```go
-// Use `ctx` and `meter` from last steps.
-
 // Create Histogram instruments
-intHistogram := metric.Must(meter).NewInt64Histogram(
-    "go_metrics.int_histogram",
-    metric.WithDescription("int_histogram description"),
-)
-floatHistogram := metric.Must(meter).NewFloat64Histogram(
-    "go_metrics.float_histogram",
-    metric.WithDescription("float_histogram description"),
+intHistogram, err := meter.Int64Histogram(
+	"histogram-meter",  // name of the metric
+    // m.WithUnit("seconds")  // add unit to the metric if you want
```

Comment:
Same comment, can you please change it to "Add a unit to the metric, if needed"?